routes/middleware/tokenValidation.js

```diff
@@ -0,0 +1,95 @@
+const jwt = require("jsonwebtoken");
+
+const MESSAGE = require("../../constant");
+const User = require("../../models/User");
+
+const checkToken = async (req, res, next) => {
+  const { ACCESS_KEY, REFRESH_KEY } = process.env;
+  const { accessToken } = req.cookies;
+  const user = req.body
+    ? await User.findOne({ email: req.body.email }).exec()
+    : await User.findById(req.params.id).exec();
+
+  if (!accessToken) {
+    const newAccessToken = jwt.sign({ email: user.email }, ACCESS_KEY, {
+      expiresIn: "1h"
+    });
+    const newRefreshToken = jwt.sign({ email: user.email }, REFRESH_KEY, {
+      expiresIn: "14days"
+    });
+
+    res.cookie("accessToken", newAccessToken, {
+      httpOnly: true
+    });
+
+    user.refreshToken = newRefreshToken;
+
+    try {
+      await user.save();
+
+      return next();
+    } catch {
+      return res.json({
+        error: {
+          message: MESSAGE.SERVER_ERROR,
+          code: 500
+        }
+      });
```

Comment:
응답을 보내준 뒤 함수를 종료하는 순서를 직관적으로 보여주기위해 아래와 같이 변경하는 건 어떨까요?? 함께 이야기해보면 좋겠습니다 :)
```suggestion
      res.json({
        error: {
          message: MESSAGE.SERVER_ERROR,
          code: 500
        }
      });
      return;
```